prehab2rehab/#635: Documentation Limesurvey setup with ParticipantPortal (#641)
MORE-Platform/more-studymanager-frontend: docs/feature/lime-survey-only-study.md/LimeSurveyOnlyStudy.md

> # <span style="color: #6E8FAC">StudyManager: LimeSurvey-Only Study</span>

Reviewer: Das ist nicht worum es geht. Das besondere ist, dass man eine Studie ohne der Verwendung der App durchführen kann. Derzeit funktioniert das nur mit LimeSurvey. Das muss aber nicht so bleiben

```suggestion
# <span style="color: #6E8FAC">Conduct a MORE Study without the App/span>
```
Ich würde das als Überschrift vorschlagen

## <span style="color: #6E8FAC">Description of the Use Case</span>

> In MORE, there is a special type of study that uses only LimeSurvey questionnaires. In this case, the mobile app is not required, because participants only need access to the LimeSurvey questionnaires assigned to them.

Reviewer: Ich würde das so schreiben:

Given a simple Study where participants only need to take a single LimeSurveys. Forcing Participants to download and install the MORE App seams like a big overhead for such a Study.

Because of that MORE support for such kind of studies a mode where Researchers can provide Participants with URLs and Code and Participants can take the surveys directly via the Browser.

---

## <span style="color: #6E8FAC">Approach and Workflow Description</span>

> To enable participation without using the mobile app, the app is replaced by a web application (Participant Portal). The Participant Portal provides a simple login process including consent handling and access to the assigned LimeSurvey questionnaires.

Reviewer: ```suggestion
For such studies participants will use the Participant Portal - a simple web application - instead of the MORE mobile app. The Participant Portal provides a simple login process including consent handling and access to the assigned LimeSurvey questionnaires.
```

> The web application is accessed via a link generated by the StudyManager. Using this link, participants can log in to the Participant Portal and complete the available LimeSurvey questionnaires.

Reviewer: Erkläre den Prozess - das ist das was die Zielgruppe dieses Artikels interessiert

Der <Role> (Study Manager, Study Administrator) erstellt im Study Manager für den Participant einen link. Er übermittelt den Link und den Code an dem Participant. Der Participant kommt mit dem Link auf das Participant Portal und authentifiziert sich da mit dem Code. Danach kann er an Umfragen der Studie teilnehmen.

> Similar to the mobile app, the web application requests a redirect link for each LimeSurvey questionnaire and forwards the participant to the corresponding LimeSurvey web interface. This redirect link is only active when the associated LimeSurvey observation schedule is currently active. Participants can then complete the questionnaire there.

Reviewer: Das ist schon wieder technisches Detail für den Reseacher genügt: 

After the authentification with the Participant Portal the Participant can complete surveys.

---

## <span style="color: #6E8FAC">Implementation</span>

### <span style="color: #6E8FAC">LimeSurvey Configuration</span>

The LimeSurvey configuration documentation is available under the [more-limesurvey repository](https://github.com/MORE-Platform/more-limesurvey). There are specific settings that are needed that lime survey works with MORE.

---

### <span style="color: #6E8FAC">StudyManager Extension</span>

> #### <span style="color: #6E8FAC">Study Configuration Extension</span>

Reviewer: ```suggestion
### <span style="color: #6E8FAC">Activate Participant Portal</span>
```
ich würde einfach diese Überschrift verwenden

To enable the use of the Participant Portal, the study configuration introduces a boolean-like configuration mechanism that allows activating access to the portal for a study.

```yaml
# StudyManagerAPI.yaml - Study Schema extension
applicationAccess:
  type: array
  description: Application types which the user has access to
  items:
    type: string
```

> The `applicationAccess` field defines which applications participants can use besides the MORE App. Currently, only "participantPortal" is supported.

Reviewer: das ist für den Forscher nicht relevant

##### <span style="color: #6E8FAC">StudyManager Configuration (UI)</span>

You'll find the setting on the Study Details page for a specific study.

![Teilnehmerportal.jpg](images/Teilnehmerportal.jpg)

#### <span style="color: #6E8FAC">Participant Access to the Participant Portal</span>

When the Participant Portal is activated, the feature can be used for the study.

##### <span style="color: #6E8FAC">Technical Extension</span>

> 1.  **Participant Portal URL generation:** The backend generates a portal URL containing the study ID and a participant reference. A UUID-based mapping table in the backend links the Participant Portal user ID to internal entities (e.g., `participantId`, `observationId`, `scheduleId`) to prevent exposure of participant data.

Reviewer: ```suggestion
1.  **Participant Portal URL generation:** The backend generates a portal URL containing the study ID and a participant reference. The participant reference replaces internal IDs for the participant, observation and schedule to not expose such information and to allow attacks by guessing such IDs
```

> 2.  **Participant Portal Login Code:** A reusable login code is generated together with the URL (similar to the mobile app login token) and validated by the Gateway during authentication on the Participant Portal..

Reviewer: ```suggestion
2.  **Participant Portal Login Code:** The login code acts as the password for the Participant to authenticate with the Participant Portal. It is similar to the login token of the mobile app, but the Participant will need to provide it every time he accesses the Participant Portal 
```

Study researchers (roles: Admin, Operator) can generate these credentials per participant from the participant list.

> 

Reviewer: Da fehlt noch, dass das versenden von URL und Code nicht Teil von MORE ist. Der Researcher ist dafür verantwortlich die Daten an den participant zu übermitteln.

##### <span style="color: #6E8FAC">Participant List Extension (UI)</span>

The participant list provides a new action button for the participant itself. This action button opens a dialog where the study researcher (Role: Admin or Operator) can access the generation process via a button. The generated URL remains stable and reusable even if observation settings change.

<figure>
  <img src="images/participant-list.jpg" alt="Participant list extension - action button">
  <figcaption>Step 1: New Action Button to open the Dialog.</figcaption>
</figure>

| | |
|:--:|:--:|
| ![Participant dialog](images/participant-dialog.jpg) | ![Generated URL and code](images/url-and-code.jpg) |
| *Step 2: Open Participant Dialog with Participant Portal section.* | *Step 2: Generated URL and code for the participant portal login.* |

---

### <span style="color: #6E8FAC">Participant Portal</span>

> The Participant Portal is a web application that provides participants access to their assigned LimeSurvey questionnaires ([Participant Portal repository](https://github.com/MORE-Platform/more-participant-portal/pull/4)).

Reviewer: ```suggestion
The Participant Portal is a web application that provides participants access to their assigned LimeSurvey questionnaires ([Participant Portal repository](https://github.com/MORE-Platform/more-participant-portal/)).
```

Reviewer: Wenn es Sinn macht den Researcher auf das Code Repo zu verlinken

#### <span style="color: #6E8FAC">Description</span>

> Participants access the portal using the generated URL and login code. After login and consent confirmation, they can view study information and a list of scheduled LimeSurvey questionnaires. Only currently active surveys are accessible via redirect links.

Reviewer: Participants access the portal using the generated URL and login code. On the first login participants are ask to confirm the consent (similar as with the mobile app). After login , they can ...

Author: i shortened most of the parts in the userDocumentation to an easy to read and understand step by step guide

> #### <span style="color: #6E8FAC">Tech Stack Used</span>

Reviewer: Das ist nicht für den Researcher. Das gehöhrt eher ins Readme des PP Repos
